backend/src/main/java/com/bakdata/conquery/util/search/TrieSearch.java

```diff
@@ -30,83 +30,111 @@
 /**
  * Trie based keyword search for autocompletion and resolving.
  * <p>
- * We store not only whole words but suffixes up to length of SUFFIX_CUTOFF to enable a sort of fuzzy and partial search with longer and compound search terms.
+ * We store whole words and their ngrams to enable a sort of fuzzy and partial search with longer and compound search terms.
  * <p>
  * The output result should be in the following order:
  * 1) Original words (in order of difference)
- * 2) Suffixes (in order of difference)
+ * 2) ngrams (in order of difference)
+ * <p>
+ *
+ * @implNote If ngramLength is Integer.MAX_VALUE, the ngram trie is empty and TrieSearch::findItems only performs a prefix search using the query keywords.
+ * If TrieSearch is instantiated with ngramLength = 0, ngramLength is set to Integer.MAX_VALUE.
  */
 @Slf4j
-@ToString(of = {"suffixCutoff", "splitPattern"})
+@ToString(of = {"ngramLength", "splitPattern"})
 public class TrieSearch<T extends Comparable<T>> {
 	/**
-	 * We saturate matches to avoid favoring very short keywords, when multiple keywords are used.
+	 * Weight of a search hit for query and keyword of different lengths.
+	 * Should be greater than one because the weights of hits in the whole worlds trie are squared.
 	 */
-	private static final double EXACT_MATCH_WEIGHT = 0.1d;
+	private static final long BASE_WEIGHT = 2;
```

Comment:
```suggestion
	@Min(2)
	private static final long BASE_WEIGHT = 2;
```

Das hat zwar keinen Effekt, liest sich aber nicht schlecht :)